raft/src/main/java/org/apache/kafka/raft/EpochState.java

```diff
@@ -59,4 +59,10 @@ default Optional<LogOffsetMetadata> highWatermark() {
      * User-friendly description of the state
      */
     String name();
+
+    @Override
+    default void close() {
+        // Since all subclasses implement the Closeable interface while none throw any IOException,
```

Comment:
It feels odd to add a default implementation just for the sake of  documentation. How about using the following style instead?
```java
    /**
     * Since all subclasses implement the Closeable interface while none throw any IOException,
     * this default implementation is provided to eliminate the need for exception handling in the close operation.
     */
    @Override
    void close();
```